src/types.rs

```diff
@@ -146,26 +147,49 @@ pub enum ClassificationReason {
 /// Use Default implementation
 impl<V> NitfField<V>
 where
-    V: FromStr + Debug,
+    V: FromStr + Debug + Default,
     <V as FromStr>::Err: Debug,
 {
     /// Read the specified number of bytes and parse the value of a given field
-    pub fn read<T: Sized + Into<u64>>(&mut self, reader: &mut File, n_bytes: T) {
+    pub fn read<T: Sized + Into<u64>>(
+        &mut self,
+        reader: &mut File,
+        n_bytes: T,
+        field_name: &str,
+    ) -> NitfResult<()> {
         self.length = n_bytes.into();
         self.bytes = vec![0; self.length as usize];
-        self.offset = reader.stream_position().unwrap();
-        reader.read_exact(&mut self.bytes).unwrap();
-        let result = String::from_utf8(self.bytes.to_vec());
-        match result {
+
+        // Crash if something goes wrong with the cursor
+        self.offset = reader
+            .stream_position()
+            .or(Err(NitfError::Fatal(field_name.to_string())))?;
+
+        // Crash if there is an error reading the bytes
+        reader
+            .read_exact(&mut self.bytes)
+            .or(Err(NitfError::Fatal(field_name.to_string())))?;
+
+        // Try to read the bytes to a string
+        match String::from_utf8(self.bytes.to_vec()) {
+            // If it's ok, trim and try to parse to enum/native representation
             Ok(str) => {
                 self.string = str.trim().to_string();
-                self.val = self.string.parse().unwrap();
+
+                // Warn and assign a default value if error parsing
+                self.val = self.string.parse().unwrap_or_else(|_| {
+                    warn!("Non-fatal error parsing {}", field_name);
+                    V::default()
+                });
             }
-            Err(err) => {
+
+            Err(_) => {
                 self.string = String::from("Error parsing string");
-                println!("{}", err)
+                warn!("Failed to parse {field_name} from bytes: {:?}", self.bytes);
             }
         }
+        trace!("{:?}", self.val);
```

Comment:
If the above `Err()` arm is hit, the `field.val` is still default, so no action is required